waiter/src/waiter/token.clj

```diff
@@ -584,25 +589,49 @@
+                          (coll? owner-param) (set owner-param)
+                          :else (list-token-owners kv-store))
+                 parameter-filter-predicates (map
+                                               (fn [[parameter-name allowed-parameter-values]]
+                                                 (fn [token-parameters]
+                                                   (and (contains? token-parameters parameter-name)
+                                                        (contains? allowed-parameter-values
+                                                                   (str (get token-parameters parameter-name))))))
+                                               (pc/map-vals
+                                                 (fn [params]
+                                                   (cond
+                                                     (string? params) #{params}
+                                                     :else (set params)))
+                                                 (-> request-params
+                                                   (select-keys sd/token-user-editable-keys)
+                                                   (dissoc "owner"))))
+                 token->token-parameters (fn [token]
+                                           (-> (sd/token->token-description kv-store token :include-deleted include-deleted)
+                                             (token-description->editable-token-parameters)))]
              (->> owners
                   (map
                     (fn [owner]
                       (->> (list-index-entries-for-owner kv-store owner)
                            (filter
-                             (fn [[_ entry]]
+                             (fn list-tokens-delete-predicate [[_ entry]]
                                (or include-deleted (not (:deleted entry)))))
                            (filter
-                             (fn [[token _]]
+                             (fn list-tokens-auth-predicate [[token _]]
                                (or (nil? can-manage-as-user)
                                    (authz/manage-token? entitlement-manager can-manage-as-user token {"owner" owner}))))
+                           (filter
+                             (fn list-tokens-parameters-predicate [[token _]]
+                               (let [token-parameters (token->token-parameters token)]
+                                 (every? #(% token-parameters) parameter-filter-predicates))))
                            (map
                              (fn [[token entry]]
                                (-> (if show-metadata
                                      (update entry :last-update-time tc/from-long)
                                      (dissoc entry :deleted :etag :last-update-time))
-                                   (assoc :owner owner :token token)))))))
+                                 (assoc :owner owner :token token)))))))
```

Comment:
I disagree with the change in indentation here.

https://guide.clojure.style/#threading-macros
https://github.com/clojure/clojure/blob/ee3553362de9bc3bfd18d4b0b3381e3483c2a34c/src/clj/clojure/core.clj#L4095-L4096

Reply:
The body indentation mechanism makes editing uses of the arrow macros easier.
E.g. changing `->` to `some->` will not introduce additional whitespace changes.

Reply:
I wouldn't have commented if this were new code, but this changed was applied to a line that was otherwise untouched in the patch. Our team just needs to decide on a standard to avoid auto format noise like this in the future.